packages/synapse-react-client/src/components/EntityFinder/tree/VirtualizedTree.tsx

```diff
@@ -523,27 +538,44 @@ export const VirtualizedTree = (props: VirtualizedTreeProps) => {
     // Because we update the root node with a copy at the end of this function, we can write to the node under update.
     async (node: Writable<EntityHeaderNode>) => {
       // Fetch the children of the node
-      const children = await SynapseClient.getEntityChildren(
-        {
-          parentId: node.id,
-          nextPageToken: node.childrenNextPageToken,
-          includeTypes: visibleTypes,
-        },
-        accessToken,
-      )
-
+      const entityChildrenRequest: EntityChildrenRequest = {
+        parentId: node.id,
+        nextPageToken: node.childrenNextPageToken,
+        includeTypes: visibleTypes,
+      }
+      const children = await queryClient.fetchQuery({
+        queryKey: keyFactory.getEntityChildrenQueryKey(
+          entityChildrenRequest,
+          false,
+        ),
+        queryFn: () => getEntityChildren(entityChildrenRequest, accessToken),
+      })
       // Update the node data -- add the children and store the nextPageToken
       if (node.children) {
-        node.children.push(...children.page)
+        for (const newChild of children.page) {
+          // Prevent adding duplicate entries
+          if (!node.children.find(child => child.id == newChild.id)) {
+            node.children.push(newChild)
+          }
+        }
```

Comment:
It's challenging to figure out how to prevent this from being called twice, since it's a complex mix of our code and `react-vtree`. It's comparatively much easier to just prevent appending a duplicate when it happens.

Reply:
Would you mind adding the ticket number to this comment and/or adding some of the context you've provided here for why checking for duplicates is necessary?